app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Filament\Resources\GameResource\RelationManagers;
+
+use App\Models\MemoryNote;
+use App\Models\User;
+use Filament\Forms;
+use Filament\Forms\Form;
+use Filament\Resources\RelationManagers\RelationManager;
+use Filament\Support\Enums\FontFamily;
+use Filament\Tables;
+use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+
+class MemoryNotesRelationManager extends RelationManager
+{
+    protected static string $relationship = 'memoryNotes';
+
+    protected static ?string $title = 'Code Notes';
+
+    public static function canViewForRecord(Model $ownerRecord, string $pageClass): bool
+    {
+        /** @var User $user */
+        $user = auth()->user();
+
+        return $user->can('manage', MemoryNote::class);
+    }
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([]);
+    }
+
+    public function table(Table $table): Table
+    {
+        $gameId = $this->ownerRecord->id;
+
+        $associatedUsers = User::whereHas('authoredCodeNotes', function (Builder $query) use ($gameId) {
+            $query->where('game_id', $gameId);
+        })->get();
+
+        return $table
+            ->recordTitleAttribute('address')
+            ->searchPlaceholder('Search (Body)')
+            ->modifyQueryUsing(fn (Builder $query) => $query->where('body', '!=', ''))
+            ->columns([
+                Tables\Columns\TextColumn::make('address_hex')
+                    ->label('Address')
+                    ->fontFamily(FontFamily::Mono)
+                    ->color('info'), // Add some visual distinction between the address and the body.
```

Comment:
![Screenshot 2024-07-21 at 4 39 13 PM](https://github.com/user-attachments/assets/deaa23d1-1b31-4e4d-b704-983bbaf69918)

I found myself getting disoriented when scrolling through long lists of notes. Changing the address color helped the problem a lot.